pkg/k8s/apis/cilium.io/v2alpha1/types.go

```diff
@@ -75,3 +76,78 @@ type CiliumEndpointSliceList struct {
 	// Items is a list of CiliumEndpointSlice.
 	Items []CiliumEndpointSlice `json:"items"`
 }
+
+// +genclient
+// +genclient:nonNamespaced
+// +genclient:onlyVerbs=get,list,watch
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:categories={cilium},singular="ciliumflowlogging",path="ciliumflowloggings",scope="Cluster",shortName={cfl,ciliumfl}
+// +kubebuilder:printcolumn:JSONPath=".metadata.creationTimestamp",description="Time duration since creation of CiliumFlow Logging",name="Age",type=date
+// +kubebuilder:storageversion
+// +kubebuilder:subresource:status
+
+// CiliumFlowLogging configures a flow logging task for Cilium agents.
+//
+// +deepequal-gen=false
+type CiliumFlowLogging struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	// Spec defines the desired specification/configuration of the flow logging.
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Spec is immutable"
+	Spec FlowLoggingSpec `json:"spec"`
+
+	// Status defines the realized specification/configuration and status
+	// of the flow logging.
+	//
+	// +kubebuilder:validation:Optional
+	Status FlowLoggingStatus `json:"status,omitempty"`
+}
+
+// FlowLoggingSpec defines configuration of a flow logging task.
+//
+// +deepequal-gen=false
```

Comment:
why not auto-generate DeepEqual?

Reply:
It is not needed (in this version it won't handle object updates at all). IIRC there were some issues when I left the default.